### PR TITLE
add request id to error meta

### PR DESCRIPTION
**Pull request by nchagrass**
## Context

- Current error handling from the Dash are currently not including a precious request-id, this should add it to the output

### CODE REVIEW

File: lib/index.js [hunk 1]
```diff
@@ -76,7 +82,13 @@ function makeRequest(method, fullUrl, body, options) {
 
 			error.code = codeStr;
 			error.statusCode = code;
-			error.meta = { httpStatus: code, method: method, url: fullUrl, data: body };
+			error.meta = { 
+				httpStatus: code,
+				method: method,
+				url: fullUrl,
+				data: body,
+				requestID: response.headers['request-id'] ?? '',
```
Comment (0xdeafcafe) [inline lib/index.js]:
```suggestion
				requestId: response.headers['request-id'] ?? '',
```

File: lib/index.js [hunk 2]
```diff
@@ -64,7 +64,13 @@ function makeRequest(method, fullUrl, body, options) {
 
 					error.code = 'invalid_json';
 					error.statusCode = code;
-					error.meta = { httpStatus: code, method: method, url: fullUrl, data: body };
+					error.meta = { 
+						httpStatus: code,
+						method: method,
+						url: fullUrl,
+						data: body,
+						requestID: response.headers['request-id'] ?? '',
```
Comment (nchagrass) [inline lib/index.js]:
```suggestion
						requestId: response.headers['request-id'] ?? '',
```

File: lib/index.js [hunk 3]
```diff
@@ -64,7 +64,13 @@ function makeRequest(method, fullUrl, body, options) {
 
 					error.code = 'invalid_json';
 					error.statusCode = code;
-					error.meta = { httpStatus: code, method: method, url: fullUrl, data: body };
+					error.meta = { 
+						httpStatus: code,
+						method: method,
+						url: fullUrl,
+						data: body,
+						requestId: response.headers['request-id'] ?? '',
```
Comment (0xdeafcafe) [inline lib/index.js]:
```suggestion
						requestId: response.headers['request-id'] ?? void 0,
```

File: lib/index.js [hunk 4]
```diff
@@ -76,7 +82,13 @@ function makeRequest(method, fullUrl, body, options) {
 
 			error.code = codeStr;
 			error.statusCode = code;
-			error.meta = { httpStatus: code, method: method, url: fullUrl, data: body };
+			error.meta = { 
+				httpStatus: code,
+				method: method,
+				url: fullUrl,
+				data: body,
+				requestId: response.headers['request-id'] ?? '',
```
Comment (0xdeafcafe) [inline lib/index.js]:
```suggestion
				requestId: response.headers['request-id'] ?? void 0,
```

**Review by 0xdeafcafe — APPROVED** [review]
Little nit pick, but it's probably better to just omit the value if not present/null.

**Comment by nchagrass** [comment]
> Little nit pick, but it's probably better to just omit the value if not present/null.

I did not know that even existed, nice one!

**Pull request merged**